client/components/Byline/Byline.js

```diff
@@ -1,76 +1,84 @@
+		// Array of authors (e.g. from pub data)
+		PropTypes.arrayOf(PropTypes.oneOfType([PropTypes.string, PropTypes.shape({})])),
+		// Byline (e.g. from ForeignPublication)
+		PropTypes.string,
+	]).isRequired,
 	bylinePrefix: PropTypes.string,
-	hideAuthors: PropTypes.bool,
-	hideContributors: PropTypes.bool,
 	linkToUsers: PropTypes.bool,
 	renderEmptyState: PropTypes.func,
 	renderSuffix: PropTypes.func,
 };
 
-const defaultProps = {
+export const defaultProps = {
 	bylinePrefix: 'by',
-	hideAuthors: false,
-	hideContributors: true,
 	linkToUsers: true,
 	renderEmptyState: () => null,
 	renderSuffix: () => null,
 };
 
 const Byline = (props) => {
-	const {
-		pubData,
-		bylinePrefix,
-		hideAuthors,
-		hideContributors,
-		linkToUsers,
-		renderEmptyState,
-		renderSuffix,
-	} = props;
-	const authors = getAllPubContributors(pubData, hideAuthors, hideContributors);
-
+	const { bylinePrefix, contributors, linkToUsers, renderEmptyState, renderSuffix } = props;
 	const renderContent = () => {
 		return (
 			<>
 				{bylinePrefix && <span>{bylinePrefix} </span>}
-				{authors.map((author, index) => {
-					const separator =
-						index === authors.length - 1 || authors.length === 2 ? '' : ', ';
-					const prefix = index === authors.length - 1 && index !== 0 ? ' and ' : '';
-					const user = author.user;
-					if (user.slug && linkToUsers) {
+				{typeof contributors === 'string' ? (
+					<span>{contributors}</span>
+				) : (
+					contributors.map((author, index) => {
+						const separator =
+							index === contributors.length - 1 || contributors.length === 2
+								? ''
+								: ', ';
+						const prefix =
+							index === contributors.length - 1 && index !== 0 ? ' and ' : '';
+
+						if (typeof author === 'string') {
+							return (
+								<span key={`author-${author}`}>
+									{prefix}
+									{author}
+									{separator}
+								</span>
+							);
+						}
+
+						const user = author.user;
+
+						if (user.slug && linkToUsers) {
+							return (
+								<span key={`author-${user.id}`}>
+									{prefix}
+									<a href={`/user/${user.slug}`} className="hoverline">
+										{user.fullName}
+									</a>
+									{separator}
+								</span>
+							);
+						}
 						return (
 							<span key={`author-${user.id}`}>
 								{prefix}
-								<a href={`/user/${user.slug}`} className="hoverline">
-									{user.fullName}
-								</a>
+								{user.fullName}
 								{separator}
 							</span>
 						);
-					}
-					return (
-						<span key={`author-${user.id}`}>
-							{prefix}
-							{user.fullName}
-							{separator}
-						</span>
-					);
-				})}
+					})
+				)}
 			</>
 		);
 	};
 
 	return (
 		<div className="byline-component byline">
 			<span className="text-wrapper">
-				{authors.length > 0 && renderContent()}
-				{authors.length === 0 && renderEmptyState()}
+				{contributors.length > 0 && renderContent()}
+				{contributors.length === 0 && renderEmptyState()}
```

Comment:
This works nicely for both the string and array forms of `props.contributors` so I didn't need to change it.